C99 loops

jhunkeler opened the pull request:
The changes are in the title.

I also fixed a segfault in `tables/c_imt.c:c_imtclose` where `imt_descr->files` could be `NULL` before accessing the array element(s).

jhunkeler committed:
Tables: Update for-loops to C99

* Removes extra scope(s)
* Changes iterator('s) type to size_t where possible
* Fix formatting where possible

jhunkeler committed:
CTEGEN2: Update for-loops to C99

* Removes extra scope(s)
* Changes iterator('s) type to size_t where possible
* Fix formatting where possible

jhunkeler committed:
WFC3: Update for-loops to C99

* Removes extra scope(s)
* Changes iterator('s) type to size_t where possible
* Fix formatting where possible

jhunkeler committed:
Core Library: Update for-loops to C99

* Removes extra scope(s)
* Changes iterator('s) type to size_t where possible
* Fix formatting where possible

jhunkeler committed:
HSTIO: Update for-loops to C99

* Removes extra scope(s)
* Changes iterator('s) type to size_t where possible
* Fix formatting where possible

jhunkeler commented on hstio/hstio.c:
```diff
@@ -277,7 +277,7 @@ int findTotalNumberOfHDUSets(const char * fileName, const char * setContainsExtN
             int exact = FALSE;
             int caseSensitive = TRUE;
-            fits_compare_str(setContainsExtName, keyValue, caseSensitive, &match, &exact);
+            fits_compare_str((char *) setContainsExtName, keyValue, caseSensitive, &match, &exact);
```
`findTotalNumberOfHDUSets` expects `setContainsExtN` to be `const char *`, but `fits_compare_str` wants a `char *`. As far as I can see, no modifications to the pointer are made in `fits_compare_str`. This just squelches the warning.

jhunkeler commented on hstio/hstio.c:
```diff
@@ -1037,28 +1027,26 @@ void setStorageOrder(SingleGroup * group, enum StorageOrder storageOrder)
     //WARNING - assumes row major storage
-    {unsigned ithRow;
 #ifdef _OPENMP
-    #pragma omp parallel for shared(output, input) private(ithRow) schedule(static)
```
Previously `ithRow` was forced private. Now it's just private: `for (size_t ithRow = 0; ...)`

pllim commented on pkg/acs/lib/acscte/dopcte-gen3.c:
```diff
@@ -592,10 +587,10 @@ static int alignAmpData(FloatTwoDArray * amp, const unsigned ampID)
+    const size_t nRows = amp->ny;
 
-    Bool allocationFail = False;
+    bool allocationFail = false;
```
I was wondering why it is Bool and not bool. I am okay to change that as Master of ACS ™️ but that should be a smaller limited scope PR, not here.

jhunkeler committed:
Rebase and redo loops that were moved

jhunkeler committed:
acs: splint_nr: Fix leak on return

jhunkeler committed:
acsinfo.c: Change loop index variable scope

jhunkeler committed:
acssect.c: Change loop index variable scope

jhunkeler committed:
addk2d.c: Change loop index variable scope

jhunkeler committed:
bin2d.c: Change loop index variable scope

jhunkeler committed:
bincoords.c: Change loop index variable scope

jhunkeler committed:
defswitch.c: Change loop index variable scope

jhunkeler committed:
detchip.c: Change loop index variable scope

jhunkeler committed:
detnsegn.c: Change loop index variable scope

jhunkeler committed:
div1d.c: Change loop index variable scope

jhunkeler committed:
dodqi.c: Change loop index variable scope

jhunkeler committed:
donoise.c: Change loop index variable scope

jhunkeler committed:
dostat.c: Initialize values

jhunkeler committed:
dostat.c: Change loop index variable scope

jhunkeler committed:
findroot.c: Change loop index variable scope

jhunkeler committed:
getccdtab.c: Change loop index variable scope

jhunkeler committed:
getccdtab.c: Map table column id strings and data and change modify them using a loop

jhunkeler committed:
blev_funcs_postsm4.c: Consolidate iterator variables

jhunkeler committed:
do2d.c: Consolidate iterator variables

* Use size_t

jhunkeler committed:
dodqi.c: Consolidate iterator variables

jhunkeler committed:
dodqi.c: Consolidate iterator variables

jhunkeler committed:
doflat.c: Consolidate iterator variables

jhunkeler committed:
dophot.c: Consolidate iterator variables

* strlen returns size_t not int

jhunkeler committed:
doshad.c: Consolidate iterator variables

jhunkeler committed:
dospot.c: Consolidate iterator variables

jhunkeler committed:
getccdtab.c: Consolidate iterator variables

jhunkeler committed:
getccdtab.c: Consolidate iterator variables

* Use size_t when appropriate
* Change variable scope when appropriate

jhunkeler committed:
hstcal_memory.c: Consolidate iterator variables

jhunkeler committed:
initmulti.c: Consolidate iterator variables

jhunkeler committed:
key.c: Consolidate iterator variables

* Use size_t

jhunkeler committed:
mkname.c: Consolidate iterator variables

* Use size_t
* Use not-equal with unsigned types

jhunkeler committed:
mkoutname.c: Consolidate iterator variables

* Use size_t when appropriate

jhunkeler committed:
mkoutname.c: Simplify FindExtn

* Add delim_check function

jhunkeler committed:
mkspt.c: Consolidate iterator variables

jhunkeler committed:
mult1d.c: Consolidate iterator variables

jhunkeler committed:
multk1d.c: Consolidate iterator variables

jhunkeler committed:
multk2d.c: Consolidate iterator variables

jhunkeler committed:
parseamps.c: Consolidate iterator variables

jhunkeler committed:
prinfo.c: Consolidate iterator variables

* Cast int to char when changing case

jhunkeler committed:
spline.c: Consolidate iterator variables

jhunkeler committed:
streqic.c: Clean up streq_ic function

* Handles NULL input

jhunkeler committed:
sub1d.c: Consolidate iterator variables

jhunkeler committed:
sub2d.c: Consolidate iterator variables

jhunkeler committed:
trim2d.c: Consolidate iterator variables

jhunkeler committed:
trlbuf.c: Consolidate iterator variables

jhunkeler committed:
ufilename.c: Consolidate iterator variables

jhunkeler committed:
unbinline.c: Consolidate iterator variables

jhunkeler committed:
Pair iterator variable with the required type

* Editor bugged out so I didn't notice warnings about mismatches

jhunkeler committed:
Revert "CTEGEN2: Update for-loops to C99"

This reverts commit 60198674491385a64b2e042560eed4d7580cad0e.

jhunkeler committed:
Only proceed when model is not NULL

jhunkeler committed:
Reformat the entire file

* Too many things jammed together
* Too many nested scopes without indentation
* Indents are inconsistent

jhunkeler committed:
Roll back change from Bool to bool

* Will handle in a different PR

jhunkeler committed:
Roll back changes to index variables in OpenMP loops

* Will handle in a different PR

jhunkeler committed:
Roll back changes to index variables in OpenMP loops

jhunkeler committed:
Fix assignment in conditional branch

jhunkeler committed:
Fix invalid pointer test

jhunkeler committed:
Fix invalid indentation in unguarded if-statement

jhunkeler committed:
Fix invalid indentation in unguarded if-statement

jhunkeler committed:
Remove unused variable: nchars

jhunkeler committed:
Add TODO to remove variable: exist

jhunkeler committed:
Fix indentation of unguarded if-statement

jhunkeler committed:
Add braces to for-loop

jhunkeler committed:
Add TODO to investigate variable: off

jhunkeler committed:
Cast hstio_error to match return type

jhunkeler committed:
Remove unused variable: weight

jhunkeler committed:
Remove unused variable: z

jhunkeler committed:
Remove unused variable: k

jhunkeler committed:
STIS cs0: c99 for-loop

jhunkeler committed:
STIS cs1: c99 for-loop

jhunkeler committed:
STIS cs1: Fix formatting in if-statement

jhunkeler committed:
STIS cs2: c99 for-loops

jhunkeler committed:
STIS cs4: c99 for-loops

jhunkeler committed:
STIS cs4: set index pointer when wavelength matches, not after

jhunkeler committed:
STIS cs6: c99 for-loops

jhunkeler committed:
STIS cs6: Fix invalid pointer comparison

jhunkeler committed:
STIS cs6: Fix indentation in unguarded if-statement

jhunkeler committed:
STIS cs7: c99 for-loops

jhunkeler committed:
STIS cs7: Fix invalid pointer comparison

jhunkeler committed:
STIS cs8: c99 for-loops

jhunkeler committed:
STIS cs11: c99-loops

jhunkeler committed:
STIS cs11: Increase history buffer size

jhunkeler committed:
STIS cs11: Fix formatting

jhunkeler committed:
STIS cs11: Use snprintf

jhunkeler committed:
STIS cs11: Fix formatting

jhunkeler committed:
STIS cs12: c99 for-loops

jhunkeler committed:
STIS idtalg: c99 for-loops

jhunkeler committed:
STIS lib: c99 for-loops

jhunkeler committed:
STIS lib: Add TODO to investigate variable: off

jhunkeler committed:
STIS lib: Change global status on error

jhunkeler committed:
STIS lib: strlen returns size_t

* Cast to int

jhunkeler committed:
STIS src: c99 for-loops

jhunkeler committed:
tables: cast strlen to int

jhunkeler committed:
tables: c99 for-loops

jhunkeler committed:
tables: cast IRAF_INDEFI to int

jhunkeler committed:
acscte: c99 for-loops

jhunkeler committed:
ACS lib: c99 for-loops

jhunkeler committed:
ACS lib: remove unused variables: npix_x, npix_y

jhunkeler committed:
ACS lib: remove unused variables: npix_x, npix_y

jhunkeler committed:
ACS lib: cast return value of strlen to int

jhunkeler committed:
ACS lib: remove unused variable 'k'

* The return value is not used meaningfully

jhunkeler committed:
WFC3 calwf3: Use variables suffix_flt and suffix_flc as intended

jhunkeler committed:
WFC3 calwf3: Fix indentation

jhunkeler committed:
WFC3 calwf3: Fix indentation

jhunkeler committed:
WFC3 calwf3: Fix indentation

jhunkeler committed:
WFC3 wf3ccd: Fix indentation

jhunkeler committed:
WFC3 wf3cte: Add TODO for unused variables

jhunkeler committed:
WFC3 wf3ccd: Fix index variable type

jhunkeler committed:
WFC3 wf3ccd: Remove unused variable: subarray

* Set but not referenced

jhunkeler committed:
WFC3 wf3ir: Add TODO for unused function arguments

* Passing indexes in and not using them feels like a bug to me

jhunkeler committed:
WFC3 wf3ir: Add TODO for variable: eval

* The error data is ignored, on purpose?

jhunkeler committed:
WFC3 wf3sum: cast strlen return value to int

jhunkeler committed:
WFC3 wf32d: Fix formatting

jhunkeler committed:
WFC3 wf32d: Fix formatting

jhunkeler committed:
WFC3 lib: Add TODO for unused variable: off

jhunkeler committed:
ACS: Update for-loops to C99

* Removes extra scope(s)
* Changes iterator('s) type to size_t where possible
* Fix formatting where possible

# Conflicts:
#	pkg/acs/lib/acsccd/blev_funcs_postsm4.c
#	pkg/acs/lib/acsccd/doccd.c

pllim commented:
Joe, you are resilient... 😆

mcara reviewed (changes requested):
I think the consistency of types needs improvement. I pointed to a couple of cases. It is not a complete list.
